foreman/data_refinery_foreman/surveyor/geo.py

```diff
@@ -359,27 +348,57 @@ def create_experiment_and_samples_from_api(self, experiment_accession_code) -> (
                         sample_object.has_raw = True
                         sample_object.save()
 
+                    # filename and source_filename are the same for these
+                    filename = supplementary_file_url.split('/')[-1]
                     original_file = OriginalFile.objects.get_or_create(
                             source_url = supplementary_file_url,
-                            source_filename = supplementary_file_url.split('/')[-1],
+                            filename = filename,
+                            source_filename = filename,
                             has_raw = sample_object.has_raw,
                             is_archive = True
                         )[0]
 
+                    logger.debug("Created OriginalFile: " + str(original_file))
+
                     original_file_sample_association = OriginalFileSampleAssociation.objects.get_or_create(
                             original_file = original_file,
                             sample = sample_object
                         )
 
+                    if original_file.is_CEL_file():
+                        # Only Affymetrix Microarrays produce .CEL files
+                        sample_object.technology = 'MICROARRAY'
+                        sample_object.manufacturer = 'AFFYMETRTIX'
+                        sample_object.save()
+
+                # It's okay to survey RNA-Seq samples from GEO, but we
+                # don't actually want to download/process any RNA-Seq
+                # data unless it comes from SRA.
+                if sample_object.technology != 'RNA-SEQ':
+                    created_samples.append(sample_object)
+
+                # Now that we've determined the technology at the
+                # sample level, we can set it at the experiment level,
+                # just gotta make sure to only do it once.
+                # XXX: Do we want to remove this field? It's not going
```

Comment:
If we're touching this code, can we address this concern now or remove the triple X? :vin_diesel: